xbmc/view/GUIViewState.cpp

```diff
@@ -344,6 +345,32 @@ void CGUIViewState::SetSortMethod(SortDescription sortDescription)
   return SetSortMethod(sortDescription.sortBy, sortDescription.sortOrder);
 }
 
+bool CGUIViewState::ChooseSortMethod()
+{
+  
+  CGUIDialogSelect *dialog = static_cast<CGUIDialogSelect *>(g_windowManager.GetWindow(WINDOW_DIALOG_SELECT));
+  if (dialog)
+  {
+    dialog->Reset();
+    dialog->SetHeading(CVariant{ 32104 }); // Label "Sort by"
+    for (int i = 0; i < m_sortMethods.size(); ++i)
+    {
+      dialog->Add(g_localizeStrings.Get(m_sortMethods[i].m_buttonLabel));
+    }
+    dialog->SetSelected(m_currentSortMethod);
+    dialog->Open();
+    int newSelected = dialog->GetSelectedItem();
+    // check if selection has changed
+    if (!dialog->IsConfirmed() || newSelected < 0 || newSelected == m_currentSortMethod)
+      return false;
+
+    m_currentSortMethod = newSelected;
+
+    SaveViewState();
+  }
+  return true;
```

Comment:
now it returns success if dialog is null and wasn't opened. probably best to swap the two return statements (such that the success case is the inner-most, failure last)